### DISCUSSION TITLE
Avalonia support (Spoiler Alert: Impossible)

### DISCUSSION THREAD

**Discussion by Haltroy** [cefsharp/CefSharp#3438] [General]
[Avalonia UI](https://avaloniaui.net/) is a cross-platform XAML framework that helps developers to create GUI applications for Linux, Windows and macOS. Avalonia uses .NET Core (and also .NET 5 and newer) and since CefSharp now supports .NET Core (and also .NET 5 and newer), is there a way to use CefSharp with Avalonia UI?
Is there a way to make CefSharp work with Avalonia so we can develop apps outside of Windows? 
I know that this just sounds like a big request but I just want to know that if it's possible or not.  A simple "yes" or "no" is enough to satisfy me.  I know you don't necessarily have to do it for everything. As I can tell (and I'm not very good at it), it is possible to make a custom control for Avalonia like in WinForms, WPF and OffScreen.
If this is a stupid question, I'm really sorry. Trying my best to not being stupid but sometimes it can slip.

Thanks.

**Comment by amaitland** [comment]
> Is there a way to make CefSharp work with Avalonia so we can develop apps outside of Windows?

CefSharp is written using Visual C++ which only runs on Windows. Microsoft have said they have no plans to port Visual C++  to other operating systems.

**Reply by Haltroy** [reply]
Thanks for the reply. I guess it's not possible then.
At first, I thought CefSharp was written in just "C++", not "Visual C++" and asked myself that if this is possible and it lead me to here.
Currently, I'm trying to find a way to use "C++" instead of "Visual C++" but that's challenging for a dumb C# programmer like me. I already noticed some limitations on my search:
 - Not using Visual C++ can cause big headaches since it will be hard to link to .NET platforms.
 - Development will be hard since debugging Visual C++ was more easy.

At this point, I thought just creating a new wrapper for CEF is more suitable. I wish that MS had a plan for this. I mean, they did .NET Core a cross-platform but their WinForms and WPF are still Windows only and Mono is now iOS and Android only (Xamarin). 
Anyways, now I either have to write my own wrapper or search all around web. If you know a different CEF wrapper that can support cross-platform, let me know.
I'm taking this answer as a "hard-as-a-diamond no".

**Reply by amaitland** [reply]
Others are using CefGlue on Linux and Mac osx, one such example project is https://github.com/chromelyapps/Chromely

**Reply by Haltroy** [reply]
Thanks, I already searched other wrappers and I decided to use CefGlue on Linux & macOS. I only need to find something for mobile (Xamarin). I hope that I won't need to search and the built-in web browser control in Xamarin can satisfy my needs (unlike .NET Framework's control, which is why people use CefSharp in the first place).
